### ISSUE TITLE
Weird file appeared alongside an edited text file ended with ~

### ISSUE THREAD

**Issue by farhatizakaria** [elementary/files#2271]
### What Happened?

Weird text file appeared alongside my own text file called finished with ~ when I saved what I wrote

### Steps to Reproduce

1. I create a text file
2. I open (Pressing enter not double click) the text file and adding/modifing the content in it
3. Save it
4. Another text file appeared alongside the original one got the same name ended with ~
5. I think that the file manager should do an autorefresh after saving a text file to not showing the same file ended with ~
### Expected Behavior

The file manager should autorefresh itself to hiding the ~ file 😐 
### OS Version

7.x (Horus)

### Software Version

Latest release (I have run all updates)

### Hardware Info

```
System:
  Host: dell-latitude Kernel: 5.19.0-50-generic x86_64 bits: 64
    Desktop: Pantheon Distro: elementary OS 7 Horus
Machine:
  Type: Laptop System: Dell product: Latitude E7450 v: N/A
    serial: <superuser required>
  Mobo: Dell model: 0D8H72 v: A00 serial: <superuser required> UEFI: Dell
    v: A24 date: 04/11/2020
Battery:
  ID-1: BAT0 charge: 24.8 Wh (73.6%) condition: 33.7/53.3 Wh (63.2%)
CPU:
  Info: dual core model: Intel Core i5-5300U bits: 64 type: MT MCP cache:
    L2: 512 KiB
  Speed (MHz): avg: 2295 min/max: 500/2900 cores: 1: 2295 2: 2295 3: 2295
    4: 2295
Graphics:
  Device-1: Intel HD Graphics 5500 driver: i915 v: kernel
  Device-2: Sunplus Innovation Integrated_Webcam_HD type: USB
    driver: uvcvideo
  Display: x11 server: X.Org v: 1.21.1.4 driver: X: loaded: modesetting
    unloaded: fbdev,vesa gpu: i915 resolution: 1366x768~60Hz
  OpenGL: renderer: Mesa Intel HD Graphics 5500 (BDW GT2)
    v: 4.6 Mesa 22.2.5-0ubuntu0.1~22.04.3
Audio:
  Device-1: Intel Broadwell-U Audio driver: snd_hda_intel
  Device-2: Intel Wildcat Point-LP High Definition Audio
    driver: snd_hda_intel
  Sound Server-1: ALSA v: k5.19.0-50-generic running: yes
  Sound Server-2: PulseAudio v: 15.99.1 running: yes
  Sound Server-3: PipeWire v: 0.3.48 running: yes
Bluetooth:
  Device-1: Intel Bluetooth wireless interface type: USB driver: btusb
  Report: hciconfig ID: hci0 rfk-id: 2 state: down
    bt-service: enabled,running rfk-block: hardware: no software: no
    address: 4C:34:88:62:0A:92
Drives:
  Local Storage: total: 465.76 GiB used: 17.62 GiB (3.8%)
  ID-1: /dev/sda vendor: Seagate model: ST500LX012-1LM162-SSHD
    size: 465.76 GiB
Partition:
  ID-1: / size: 92.95 GiB used: 17.6 GiB (18.9%) fs: ext4 dev: /dev/sda8
  ID-2: /boot/efi size: 1022 MiB used: 15.5 MiB (1.5%) fs: vfat
    dev: /dev/sda6
Swap:
  ID-1: swap-1 type: partition size: 4 GiB used: 0 KiB (0.0%) dev: /dev/dm-0
Sensors:
  System Temperatures: cpu: 53.0 C mobo: 41.0 C sodimm: SODIMM C
  Fan Speeds (RPM): cpu: 0
Info:
  Processes: 263 Uptime: 31m Memory: 7.63 GiB used: 2.42 GiB (31.6%)
  Shell: fish inxi: 3.3.13
```

**Comment by jeremypw** [comment]
Which app did you use to edit the text file? Elementary Code creates this backup file when the text is edited but does not delete it until the document is closed.  So while the file exists it is not an error for Files to show it.  You can avoid seeing this file by turning "Show Hidden files" off, either through the app menu or by pressing `<Ctrl>H`.

**Comment by farhatizakaria** [comment]
@jeremypw even I disable "Show Hidden files" it still show itself not as other distros, also when I create a text file and rename it then I press enter, also another enter press to enter the file, it just creates another "New text file" and open "Code", so I figure out that the main reason for this issues is that the file manager does not refresh itself for a create or rename a file (especially text files)

**Comment by jeremypw** [comment]
OK, if you are using the unstable version of Files this might be the issue that is fixed by #2260.  Does the unexpected file disappear if you press `F5` or the refresh button?

**Comment by farhatizakaria** [comment]
@jeremypw yes it disappear, but not automatically after creation a file, rename it or edit it

**Comment by jeremypw** [comment]
OK, thanks. Could you just confirm which version of Files you are using? If you installed it from AppCenter then you can get the information from the Files page on that app. Or did you compile from the the current `main` branch? 

I have tried to reproduce with Files v 6.4.1 but I find to that Code backup file disappears from Files automatically when the document or Code is closed. Creating/deleting files from the terminal also causes Files to refresh as expected.  There is an issue with resorting the view after such operations that is being fixed by #2260.

**Comment by jeremypw** [comment]
Just found the cause of this (and why I could not reproduce it.  It only happens if "Show Hidden Files" is "OFF" - I had it "ON".  Will fix soon.
